src/Service/Indexer/SiteKit/DefaultSchema2xDocumentEnricher.php

```diff
@@ -0,0 +1,457 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atoolo\Search\Service\Indexer\SiteKit;
+
+use Atoolo\Resource\Loader\SiteKitNavigationHierarchyLoader;
+use Atoolo\Resource\Resource;
+use Atoolo\Search\Exception\DocumentEnrichingException;
+use Atoolo\Search\Service\Indexer\ContentCollector;
+use Atoolo\Search\Service\Indexer\DocumentEnricher;
+use Atoolo\Search\Service\Indexer\IndexDocument;
+use Atoolo\Search\Service\Indexer\IndexSchema2xDocument;
+use DateTime;
+use Exception;
+
+/**
+ * @phpstan-type Phone array{
+ *     countryCode?:string,
+ *     areaCode?:string,
+ *     localNumber?:string
+ * }
+ * @phpstan-type PhoneData array{phone:Phone}
+ * @phpstan-type PhoneList array<PhoneData>
+ * @phpstan-type Email array{email:string}
+ * @phpstan-type EmailList array<Email>
+ * @phpstan-type ContactData array{
+ *     phoneList?:PhoneList,
+ *     emailList:EmailList
+ * }
+ * @phpstan-type AddressData array{
+ *     buildingName?:string,
+ *     street?:string,
+ *     postOfficeBoxData?: array{
+ *          buildingName?:string
+ *     }
+ * }
+ * @phpstan-type ContactPoint array{
+ *     contactData?:ContactData,
+ *     addressData?:AddressData
+ * }
+ * @implements DocumentEnricher<IndexSchema2xDocument>
+ */
+class DefaultSchema2xDocumentEnricher implements DocumentEnricher
+{
+    public function __construct(
+        private readonly SiteKitNavigationHierarchyLoader $navigationLoader,
+        private readonly ContentCollector $contentCollector
+    ) {
+    }
+
+    public function isIndexable(Resource $resource): bool
+    {
+        $noIndex = $resource->getData()->getBool('init.noIndex');
+        return $noIndex !== true;
+    }
+
+    /**
+     * @throws DocumentEnrichingException
+     */
+    public function enrichDocument(
```

Comment:
it might make sense for this huge method to do `$data = $resource->getData()` to reduce the amount of code a bit. Maybe one could even do
```php
$init = new DataBag($data->getAssociativeArray('init'));
$base = new DataBag($data->getAssociativeArray('base'));
$meta = new DataBag($data->getAssociativeArray('metadata'));
```